src/main/java/com/exasol/containers/status/ContainerStatus.java

```diff
@@ -0,0 +1,111 @@
+package com.exasol.containers.status;
+
+import java.io.Serializable;
+import java.util.EnumMap;
+import java.util.Map;
+
+import javax.annotation.processing.Generated;
+
+import com.exasol.containers.ExasolService;
+
+/**
+ * State information about the Exasol docker container.
+ */
+public final class ContainerStatus implements Serializable {
+    private static final long serialVersionUID = 5014025686797381100L;
+    private final String containerId;
+    private final Map<ExasolService, ServiceStatus> serviceStatuses = new EnumMap<>(ExasolService.class);
+
+    private ContainerStatus(final String containerId) {
+        this.containerId = containerId;
+        for (final ExasolService service : ExasolService.values()) {
+            this.serviceStatuses.put(service, ServiceStatus.NOT_CHECKED);
+        }
+    }
+
+    /**
+     * Create container state for the container with the given ID.
+     *
+     * @param containerId docker container ID
+     * @return container state
+     */
+    public static ContainerStatus create(final String containerId) {
+        return new ContainerStatus(containerId);
+    }
+
+    /**
+     * Get the ID of the container this status belongs to.
+     *
+     * @return container ID
+     */
+    public String getContainerId() {
+        return this.containerId;
+    }
+
+    /**
+     * Get the status of the given service.
+     *
+     * @param service service for which the status should be provided
+     * @return status of the service
+     */
+    public ServiceStatus getServiceStatus(final ExasolService service) {
+        return this.serviceStatuses.get(service);
+    }
+
+    /**
+     * Check if a service is ready.
+     *
+     * @param service service to be checked
+     * @return {@code true} if the service is ready
+     */
+    public boolean isServiceReady(final ExasolService service) {
```

Comment:
how about `test.getSeriveStatus(service).isReady()`

Reply:
it's just an idea. I'm not sure if it's more readable

Reply:
pr maybe `test.getSeriveStatus(service).is(READY)` where `is` == `equals`

Reply:
It's a public method and I did not want to break the existing interface. Otherwise I would have removed it completely.